core/harambe_core/parser/type_currency.py

```diff
@@ -1,70 +1,38 @@
 from typing_extensions import Annotated
+from harambe_core.parser.type_price import ParserTypePrice
+from harambe_core.parser.currencies import all_currencies, price_not_available_phrases
 
-price_not_available_phrases = [
-    "price not available",
-    "unavailable price",
-    "price upon request",
-    "contact for price",
-    "request a quote",
-    "call for price",
-    "check price in store",
-    "price tbd (to be determined)",
-    "price not disclosed",
-    "out of stock",
-    "sold out",
-    "pricing not provided",
-    "not priced",
-    "currently unavailable",
-    "n/a (not available)",
-    "ask for pricing",
-    "see details for price",
-    "price coming soon",
-    "temporarily unavailable",
-    "price hidden",
-    "tdb",
-    "n/a",
-]
-
-
-class ParserTypeCurrency:
-    def __new__(cls) -> Any:
-        return Annotated[float | None, BeforeValidator(cls.validate_currency)]
 
+class CurrencyParser:
     @staticmethod
-    def validate_currency(value: str) -> float | None:
+    def parse_currency_amount(
+        value: Union[str, float, int, None],
+    ) -> Union[float, None]:
+        return ParserTypePrice.validate_price(value)
+
+    def __call__(
+        self, value: Union[str, float, int, None]
+    ) -> dict[str, Optional[Union[str, float]]]:
         if isinstance(value, (float, int)):
-            return float(value)
-
-        value = str(value).strip()
-
-        if value.lower() in price_not_available_phrases:
-            return None
-
-        cleaned_value = re.sub(r"[^\d.,-]", "", value)
-        cleaned_value = re.sub(r"^0+(?!$)", "", cleaned_value)
-
-        if "." in cleaned_value:
-            decimal_parts = cleaned_value.split(".")
-            if len(decimal_parts[-1]) == 3:  # thousands separator issue
-                cleaned_value = cleaned_value.replace(".", "")
-
-        if cleaned_value.startswith("."):
-            cleaned_value = "0" + cleaned_value
-            return float(cleaned_value)
-
-        if "," in cleaned_value and "." in cleaned_value:
-            if cleaned_value.index(",") < cleaned_value.index("."):
-                cleaned_value = cleaned_value.replace(",", "")
-            else:
-                cleaned_value = cleaned_value.replace(".", "").replace(",", ".")
-        elif "," in cleaned_value and "." not in cleaned_value:
-            if (
-                len(cleaned_value.split(",")[-1]) != 3
-            ):  # check Ambiguous values 123,45 and 123,456
-                raise ValueError("Invalid price")
-            cleaned_value = cleaned_value.replace(",", "")
-
-        return float(cleaned_value.strip())
+            return {"currency": None, "currency_symbol": None, "amount": float(value)}
+        symbol = currency_name = amount = None
+        if value is not None:
+            value = str(value).strip()
+            if value.lower() in price_not_available_phrases:
+                return {"currency": None, "currency_symbol": None, "amount": None}
```

Comment:
Dont we want this to be NONE instead of a dict with all nones? Otherwise it fail the all null tests